SonarQube.TeamBuild.Integration/TrxFileReader.cs

```diff
@@ -151,35 +151,48 @@ private static string TryGetCoverageFilePath(string trxFilePath, ILogger logger)
         {
             Debug.Assert(File.Exists(trxFilePath));
 
-            string coverageFilePath = null;
+            if (!TryExtractCoverageFilePaths(trxFilePath, logger, out var attachmentUris))
+            {
+                return null;
+            }
 
-            if (TryExtractCoverageFilePaths(trxFilePath, logger, out IEnumerable<string> attachmentUris))
+            switch (attachmentUris.Count())
             {
-                switch (attachmentUris.Count())
-                {
-                    case 0:
-                        logger.LogDebug(Resources.TRX_DIAG_NoCodeCoverageInfo);
-                        break;
+                case 0:
+                    logger.LogDebug(Resources.TRX_DIAG_NoCodeCoverageInfo);
+                    return null;
 
-                    case 1:
-                        coverageFilePath = attachmentUris.First();
-                        logger.LogDebug(Resources.TRX_DIAG_SingleCodeCoverageAttachmentFound, coverageFilePath);
+                case 1:
+                    var attachmentName = attachmentUris.First();
+                    logger.LogDebug(Resources.TRX_DIAG_SingleCodeCoverageAttachmentFound, attachmentName);
 
-                        if (!Path.IsPathRooted(coverageFilePath))
-                        {
-                            coverageFilePath = Path.Combine(Path.GetDirectoryName(trxFilePath), Path.GetFileNameWithoutExtension(trxFilePath), "In", coverageFilePath);
-                            logger.LogDebug(Resources.TRX_DIAG_AbsoluteTrxPath, coverageFilePath);
-                        }
+                    var trxDirectoryName = Path.GetDirectoryName(trxFilePath);
+                    var trxFileName = Path.GetFileNameWithoutExtension(trxFilePath);
 
-                        break;
+                    var possibleCoveragePath = new[]
+                    {
+                        attachmentName,
+                        Path.Combine(trxDirectoryName, trxFileName, "In", attachmentName),
+                        // https://jira.sonarsource.com/browse/SONARMSBRU-361
+                        // With VSTest task the coverage file name uses underscore instead of spaces.
+                        Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), "In", attachmentName)
+                    }.FirstOrDefault(path => File.Exists(path));
```

Comment:
The FirstOrDefault is not very visible, I did not see it in my first read and the next if made me scratch my head - how can a freshly declared array be null?! Perhaps move on a separate line?